### ISSUE TITLE
JWT refresh token when a call is triggered after the expiry

### ISSUE THREAD

**Issue by mkharibalaji** [feathersjs/feathers#546]
Hi Guys,
       Is there a way to refresh the featherjs JWT token from the client side automatically if it expires untill he is explicitly logged out.
For Example if a call is triggered form the client after the expiry of the token,will the client handle the reauthentication process and trigger the call once again ?
```
const errorHandler = error => {
  app.authenticate({
    strategy: 'local',
    email: 'admin@feathersjs.com',
    password: 'admin'
  }).then(response => {
    // You are now authenticated again
  });
};
// Handle when auth fails during a reconnect or a transport upgrade
app.on('reauthentication-error', errorHandler)
```
I don't want to store the credentials on the client side in order to reauthenticate....Is there something else that could be done ?

**Comment by mkharibalaji** [comment]
@daffl out of curiosity - any idea on the milestone when this would be implemented as i am able to see the notes in the feathers-authentication client code...

... // create accessToken
      // TODO (EK): Support refresh tokens
      // TODO (EK): This should likely be a hook
      // TODO (EK): This service can be datastore backed to support blacklists :)

**Comment by marshallswain** [comment]
The coming milestone.  No timeline, yet.

**Comment by knappdev** [comment]
Waiting on this feature. would help a lot in the client side

**Comment by daffl** [comment]
Closing as a duplicate of https://github.com/feathersjs/authentication/issues/22
